### PR TITLE
Introduce universal bootstrap method for tests with autoMinMasterNodes=false

### PR DESCRIPTION
**Pull request by andrershov**
Currently, there are a few tests that use autoMinMasterNodes=false and
hence override `addExtraClusterBootstrapSettings`, mostly this is 10-30
lines of codes that are copy-pasted from class to class.

This PR introduces `bootstrapMasterNodeWithSpecifiedId` which is
suitable for all classes and copy-paste could be removed. 

Removing code is always a good thing!

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: Can we move this entirely within `InternalTestCluster`?

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: `s/previously started/existing and new/`?

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: I think this is never overridden, so can be inlined.

**Review by DaveCTurner — COMMENTED** [review]
Great, the duplication irked me too.

I think it can be further simplified, moving this whole mechanism within `InternalTestCluster`. I also wonder if we should reset `bootstrapMasterNodeId` to `-1` after use, and should assert that it is `-1` if `autoManageMinMasterNodes` is set.

Also there's a few places where you org.elasticsearch.use org.elasticsearch.fully-qualified org.elasticsearch.names, mirroring existing usage. I think we can tidy that up, at least in the places they occur here. There's only one `NODE_NAME_SETTING` worth talking about.

**Comment by andrershov** [comment]
@DaveCTurner thanks for your review!
And thank you for the suggestion to move bootstrap logic to `InternalTestCluster`, after this change `addExtraClusterBootrstrap` method could be removed from `NodeConfigurationSource` and `ESIntegTestCase`. If in the future, we will require a method to add some settings before nodes are started, it can easily be returned back.
I've made all the requested changes and I think it makes sense to have the second review round.

**Comment by andrershov** [comment]
I've missed that `InternalTestClusterTests` also overrides `addExtraClusterBootstrapSettings`. Now it can happily go away.

**Review by DaveCTurner — APPROVED** [review]
LGTM, nice work

**Pull request merged**
